functions/anonymous-functions.md

```diff
@@ -0,0 +1,198 @@
+| `userIds.map( ... );`                               | Just like `forEach`, we want to call the `map` function on our array, `userIds`.                                                                                           |
+| `userId => { ... }`                                 | `map` takes in a function. Here is an anonymous arrow function with one parameter, `userId`. `userId` will refer to each item in `userIds` as `map` goes through it.       |
+| `return`                                            | **Unlike** the `forEach` loop, **this function must `return` the value that we want** in our new array.                                                                    |
+
+### `forEach` vs. `map`
+
+The `forEach` function is useful when we want to execute an identical action for each item. `map` is useful when we want a result array that is the same length of the original array, and each item in the result array is based on a parallel item in the original array.
+
+## Uses for Anonymous Functions: `reduce`
+
+Let's look at one more useful function, `reduce`. This function has similarities to `forEach` and `map`, but this time we'll practice using the vocabulary _callback function_.
+
+The responsibility of `reduce` is to look through an array and apply some operation to eventually _reduce_ the array into _one value_.
+
+Let's return to our code that calculated our average cash tip, but let's focus on how we get the _sum_ of our tips.
+
+```javascript
+const cashTips = [4, 7, 9, 12, 3, 18, 6];
+let sum = 0;
+
+cashTips.forEach((tip) => {
+  sum += tip;
+});
+```
+
+Our current logic for getting the sum of all tips is:
+
+1. Initialize a `sum` variable at `0`
+1. Iterate through each `tip` in `cashTips`
+1. Increment `sum` by `tip`
+
+Essentially, we're _reducing_ our `cashTips` array into _one value_. This is a great time to refactor!
+
+The `reduce` function takes in a function. _This_ function takes in _two_ arguments:
+
+1. An argument to represent the accumulated value. This will replace our `sum` variable.
+1. One `item` from our array.
+
+<!-- prettier-ignore-start -->
+```js
+const cashTips = [4, 7, 9, 12, 3, 18, 6];
+let sum = 0;
+
+const ultimateTotal = cashTips.reduce((total, tip) => {
+  return total + tip;
+});
+```
+<!-- prettier-ignore-end -->
+
+| <div style="min-width:200px;"> Piece of Code </div> | Notes                                                                                                                                                    |
+| --------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `const ultimateTotal =`                             | The `reduce` function on the right-hand side of the `=` will become one value. Let's keep that single value in a `const` variable named `ultimateTotal`. |
```

Comment:
use sum for the variable
the function doesn't _become_ one value, it _evaluates_ to one value
```suggestion
| `const sum =`                             | The `reduce` function on the right-hand side of the `=` will evaluate to one value. Let's keep that single value in a `const` variable named `sum`. |
```